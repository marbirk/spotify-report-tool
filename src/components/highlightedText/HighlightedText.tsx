import React from 'react'

interface HighlightedTextProps {
    children: React.ReactNode
}

const HighlightedText = ({ children, ...props }: HighlightedTextProps) => (
    <div
        className="pl-8 pr-8 mt-8 mb-8 border-l-4 border-highlight-600 dark:border-highlight-300 italic"
        {...props}
    >
        {children}
    </div>
)

export default HighlightedText
