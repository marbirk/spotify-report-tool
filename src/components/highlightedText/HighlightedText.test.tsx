import React from 'react'
import { render, screen } from '@testing-library/react'

import HighlightedText from './HighlightedText'

describe('HighlightedText', () => {
    test('renders correctly', () => {
        render(<HighlightedText>Highlighted Text</HighlightedText>)
        const component = screen.queryByText('Highlighted Text')
        expect(component).toBeInTheDocument()
        expect(component).toMatchSnapshot()
    })
})
