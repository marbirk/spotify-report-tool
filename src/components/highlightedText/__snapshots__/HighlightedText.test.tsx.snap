// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightedText renders correctly 1`] = `
<div
  class="pl-8 pr-8 mt-8 mb-8 border-l-4 border-highlight-600 dark:border-highlight-300 italic"
>
  Highlighted Text
</div>
`;
